src/hooks/posthog/useCapturePageview.ts

```diff
@@ -0,0 +1,12 @@
+import { useEffect } from "react"
+import { useLocation } from "react-router-dom"
+import * as posthog from "../../posthog"
+import { featureFlags } from "../../constants"
+
+export const useCapturePageview = () => {
+  const location = useLocation()
+
+  useEffect(() => {
+    if (featureFlags.POSTHOG) posthog.capturePageview()
+  }, [location])
```

Comment:
I think we should specify the property of the object in the dependency array instead of passing the whole object. We only need pathname so I think that should work:

```suggestion
  useEffect(() => {
    if (featureFlags.POSTHOG) posthog.capturePageview()
  }, [location.pathname])
```

Reply:
Yeah I was thinking about it but it only returns the url w/o `?` and `#` see https://github.com/remix-run/history/blob/main/docs/api-reference.md#location. So if a path will be `<url>#something` or `<url>?query` this hook skip it and does not register the url change. Currently, we don't use the `#` and `?` in our dapp but maybe someday we will and probably we forget to update this hook. So not sure in which direction we should go 🤷